### PR TITLE
Fixes: Get minutes offset from start of the day over days.utc()

### PR DESCRIPTION
**Pull request by manjufy**
## What does this PR do?

Fixes # 

- https://github.com/calendso/calendso/issues/1427
- https://github.com/calendso/calendso/issues/1437

Looks like the changes which were merged in this PR https://github.com/calendso/calendso/pull/1323 causing to have a different behaviour for selecting slots for the next active day in a calendar.

Ex: My Calendar is at UTC +8 Asia/Kuala_Lumpur timezone.
If my availability times are from 9am to 5pm,
Calendar slots works well for the same day, however, for the next day, calendar adds 16 hours and slots always starts from 4pm next day. There is no issues with rest of the days. See for screenshots https://github.com/calendso/calendso/issues/1427

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How should this be tested?

- See the steps to reproduce  https://github.com/calendso/calendso/issues/1427 and https://github.com/calendso/calendso/issues/1437

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code and corrected any misspellings
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by emrysal** [comment]
Heya @manjufy - Thanks for this!

Could you have a look at the test/lib/slots.test.ts in context of this PR and check if the fail is due to the test being wrong; or the fix?

**Comment by manjufy** [comment]
> Heya @manjufy - Thanks for this!
> 
> Could you have a look at the test/lib/slots.test.ts in the context of this PR and check if they fail is due to the test being wrong; or the fix?

Oh. Forgot to look into tests. I will have a look.

File: lib/slots.ts [hunk 1]
```diff
@@ -17,11 +17,11 @@ export type GetSlots = {
   minimumBookingNotice: number;
 };
 
-const getMinuteOffset = (date: Dayjs, step: number) => {
```
Comment (manjufy) [inline lib/slots.ts:20]: I have renamed `step` to `frequence` for better clarity, same as `getMinuteOffset(inviteeDate, frequency)`

File: lib/slots.ts [hunk 2]
```diff
@@ -17,11 +17,11 @@ export type GetSlots = {
...
+const getMinuteOffset = (date: Dayjs, frequency: number) => {
   // Diffs the current time with the given date and iff same day; (handled by 1440) - return difference; otherwise 0
-  const minuteOffset = Math.min(date.diff(dayjs.utc().startOf("day"), "minute"), 1440) % 1440;
+  const minuteOffset = Math.min(date.diff(dayjs().utc(), "minute"), 1440) % 1440;
   // round down to nearest step
```
Comment (manjufy) [inline lib/slots.ts:23]: Thanks @emrysal  for pointing out to check the tests.

The main issue here was using the date with utc together with `.startOf`
Here is my observation. I'm at UTC +8 (Asia/Kuala_Lumpur). When you use the `date.utc().start("day")` date would look like `2022-01-13 00:08:00 +8`, that's basically doubles the minutesOffset value from 480 (8 hours *60) to 960. This affects the time slots lists for the following day. Hence the above change.

**Review by emrysal — APPROVED** [review]
Tests prove it works, looks like a solid fix. Many thanks for this; I know it's "just" one line of code but as I wrote it I know how much time these lines of code can take to debug.

**Comment by manjufy** [comment]
> Tests prove it works, looks like a solid fix. Many thanks for this; I know it's "just" one line of code but as I wrote it I know how much time these lines of code can take to debug.

🙏🏿

**Pull request merged**